cordova-fetch/index.js

```diff
@@ -0,0 +1,226 @@
+/**
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ 'License'); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ */
+
+var Q = require('q');
+var shell = require('shelljs');
+var superspawn = require('cordova-common').superspawn;
+var events = require('cordova-common').events;
+var depls = require('dependency-ls');
+var path = require('path');
+var fs = require('fs');
+var CordovaError = require('cordova-common').CordovaError;
+var isUrl = require('is-url');
+
+/* 
+ * A module that npm installs a module from npm or a git url
+ *
+ * @param {String} target   the packageID or git url
+ * @param {String} dest     destination of where to install the module
+ * @param {Object} opts     [opts={save:true}] options to pass to fetch module
+ *
+ * @return {String||Promise}    Returns string of the absolute path to the installed module.
+ *
+ */
+module.exports = function(target, dest, opts) {
+    var fetchArgs = ['install'];
+    opts = opts || {};
+    var tree1;
+
+    if(dest && target) {
+        //add target to fetchArgs Array
+        fetchArgs.push(target);
+        
+        //append node_modules to dest if it doesn't come included
+        if (path.basename(dest) !== 'node_modules') {
+            dest = path.resolve(path.join(dest, 'node_modules'));
+        }
+        
+        //create dest if it doesn't exist
+        if(!fs.existsSync(dest)) {
+            shell.mkdir('-p', dest);         
+        } 
+
+    } else return Q.reject(new CordovaError('Need to supply a target and destination'));
+
+    //set the directory where npm install will be run
+    opts.cwd = dest;
+
+    //if user added --save flag, pass it to npm install command
+    if(opts.save) {
+        events.emit('verbose', 'saving');
+        fetchArgs.push('--save'); 
+    } 
+    
+    //check if npm is installed
+    isNpmInstalled();
+
+    //Grab json object of installed modules before npm install
+    return depls(dest)
+    .then(function(depTree) {
+        tree1 = depTree;
+
+        //install new module
+        return superspawn.spawn('npm', fetchArgs, opts);
```

Comment:
Perhaps it is worth to add `--production` flag to avoid installing `devDependencies`